File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.cs
string[] input = Console.ReadLine().Split(' ');
int N = int.Parse(input[0]);
int M = int.Parse(input[1]);

int[] bascket = new int[N];
for (int i = 0; i < N; i++)
{
    bascket[i] = i + 1;
}

for (int times = 0; times < M; times++)
{
    string[] range = Console.ReadLine().Split(' ');
    int i = int.Parse(range[0]) - 1;
    int j = int.Parse(range[1]) - 1;

    while (i < j)
    {
        int temp = bascket[i];
        bascket[i] = bascket[j];
        bascket[j] = temp;
        i++;
        j--;
    }
}

Console.WriteLine(string.Join(" ", bascket));

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.c
#include <stdio.h>

int main(void) {
   int N, M;
   int arr[101] = {0};
   int i, j;
   int temp;

   scanf("%d %d", &N, &M);

   for(int num = 0; num < N; num ++) {
      arr[num] = num + 1;
   }

   for(int times = 0; times < M; times ++) {
      scanf("%d %d", &i, &j);
      for(int range = i - 1; range < j; range ++) {
         temp = arr[range];
         arr[range] = arr[j - 1];
         arr[j - 1] = temp;
         j --;
      }
   }
   for(int printnum = 0; printnum < N; printnum ++) {
      printf("%d ", arr[printnum]);
   }

   return 0;
}